// gatsby-plugin-react-redux/gatsby-ssr.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.onRenderBody = exports.wrapRootElement = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _reactRedux = require("react-redux");

var _serializeJavascript = _interopRequireDefault(require("serialize-javascript"));

var _createStore = _interopRequireDefault(require("./.tmp/createStore"));

var _constants = require("./constants");

var pageStores = new Map();

var wrapRootElement = function wrapRootElement(_ref) {
  var element = _ref.element,
      pathname = _ref.pathname;
  var store = (0, _createStore.default)();
  pageStores.set(pathname, store);
  return /*#__PURE__*/_react.default.createElement(_reactRedux.Provider, {
    store: store
  }, element);
};

exports.wrapRootElement = wrapRootElement;

var onRenderBody = function onRenderBody(_ref2, pluginOptions) {
  var setHeadComponents = _ref2.setHeadComponents,
      pathname = _ref2.pathname;

  if (pluginOptions === void 0) {
    pluginOptions = {};
  }

  var pageStore = pageStores.get(pathname);
  if (!pageStore) return;
  pageStores.delete(pathname);
  var options = (0, _extends2.default)({}, _constants.DEFAULT_OPTIONS, pluginOptions, {
    serialize: (0, _extends2.default)({}, _constants.DEFAULT_OPTIONS.serialize, pluginOptions.serialize)
  });
  var serializedState = (0, _serializeJavascript.default)(pageStore.getState(), options.serialize).replace(/'/g, "\\'"); // escape single quotes inside because we wrap it in them

  var isJSON = options.serialize.isJSON;
  setHeadComponents([/*#__PURE__*/_react.default.createElement("script", {
    key: "redux-state",
    id: _constants.SCRIPT_ELEMENT_ID,
    dangerouslySetInnerHTML: {
      __html: "window['" + options.windowKey + "'] = " + (isJSON ? "JSON.parse('" + serializedState + "')" : "eval('(" + serializedState + ")')")
    }
  })]);
};

exports.onRenderBody = onRenderBody;
